packages/teams-js/test/public/webStorage.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { Utils } from '../utils';
+
+describe('webStorage', () => {
+  const mobilePlatformMock = new FramelessPostMocks();
+  const desktopPlatformMock = new Utils();
+
+  beforeEach(() => {
+    mobilePlatformMock.messages = [];
+
+    // Set a mock window for testing
+    app._initialize(mobilePlatformMock.mockWindow);
+  });
+
+  afterEach(() => {
+    // Reset the object since it's a singleton
+    if (app._uninitialize) {
+      desktopPlatformMock.setRuntimeConfig(_minRuntimeConfigToUninitialize);
+      mobilePlatformMock.setRuntimeConfig(_minRuntimeConfigToUninitialize);
+      app._uninitialize();
+    }
+    jest.clearAllMocks();
+  });
+
+  describe('isWebStorageClearedOnUserLogOut', () => {
+    it('should not allow calls before initialization', () => {
+      expect(webStorage.isWebStorageClearedOnUserLogOut).toThrowError('The library has not yet been initialized');
+    });
+    Object.values(FrameContexts).forEach(context => {
+      it('should always allow isWebStorageClearedOnUserLogOut calls in desktop for all contexts', async () => {
+        await desktopPlatformMock.initializeWithContext(context, HostClientType.desktop);
+        const result = webStorage.isWebStorageClearedOnUserLogOut();
+        expect(result).toBeTruthy();
+      });
+    });
+    Object.values(FrameContexts).forEach(context => {
+      it('should never allow isWebStorageClearedOnUserLogOut calls in web for all contexts', async () => {
+        await desktopPlatformMock.initializeWithContext(FrameContexts.content, HostClientType.web);
+        const result = webStorage.isWebStorageClearedOnUserLogOut();
+        expect(result).not.toBeTruthy();
+      });
+    });
+    Object.values(FrameContexts).forEach(context => {
+      it('should allow calls isWebStorageClearedOnUserLogOut calls in mobile for all contexts', async () => {
+          await mobilePlatformMock.initializeWithContext(context);
+          mobilePlatformMock.setRuntimeConfig({ apiVersion: 1, supports: {webStorage: {}}});
+          const result = webStorage.isWebStorageClearedOnUserLogOut();
+          expect(result).toBeTruthy();
+      });
+    });
+  });
+
+  describe('Framed - isSupported', () => {
+    it('webStorage.isSupported should return false if the runtime says webStorage is not supported', () => {
+      desktopPlatformMock.setRuntimeConfig({ apiVersion: 1, supports: {} });
+      expect(webStorage.isSupported()).not.toBeTruthy();
+    });
+
+    it('webStorage.isSupported should return true if the runtime says webStorage is supported', () => {
+      desktopPlatformMock.setRuntimeConfig({ apiVersion: 1, supports: {webStorage: {}}});
+      expect(webStorage.isSupported()).toBeTruthy();
+    });
+  });
+  describe('Frameless - isSupported', () => {
+    it('webStorage.isSupported should return false if the runtime says webStorage is not supported', async () => {
+      await mobilePlatformMock.initializeWithContext(FrameContexts.task, HostClientType.ios);
```

Comment:
Why in one it is initialized with context and previous we are not? Plus if we are using context then we need it for all of the context not only `task` context. Same goes for `HostClientType.ios`